casper/src/test/scala/coop/rchain/casper/ValidateTest.scala

```diff
@@ -0,0 +1,186 @@
+
+    Validate.blockSignature[Id](block0) should be(false)
+    log.warns.last.contains(s"signature algorithm $unknownAlgorithm is unsupported") should be(true)
+
+    Validate.blockSignature[Id](block1) should be(false)
+    log.warns.last.contains(s"signature algorithm $rsa is unsupported") should be(true)
+  }
+
+  it should "return false on invalid ed25519 signatures" in {
+
+    implicit val chain   = createChain[StateWithChain](6).runS(initState).value
+    implicit val (sk, _) = Ed25519.newKeyPair
+    val (_, wrongPk)     = Ed25519.newKeyPair
+    val empty            = ByteString.EMPTY
+    val invalidKey       = ByteString.copyFrom(Base16.decode("abcdef1234567890"))
+    val block0           = signedBlock(0).withSender(empty)
+    val block1           = signedBlock(1).withSender(invalidKey)
+    val block2           = signedBlock(2).withSender(ByteString.copyFrom(wrongPk))
+    val block3           = signedBlock(3).withSig(empty)
+    val block4           = signedBlock(4).withSig(invalidKey)
+    val block5           = signedBlock(5).withSig(block0.sig) //wrong sig
+
+    val blocks = Vector(block0, block1, block2, block3, block4, block5)
+
+    blocks.exists(Validate.blockSignature[Id]) should be(false)
+    log.warns.size should be(blocks.length)
+    log.warns.forall(_.contains("signature is invalid")) should be(true)
+  }
+
+  it should "return true on valid ed25519 signatures" in {
+    val n                = 6
+    implicit val chain   = createChain[StateWithChain](n).runS(initState).value
+    implicit val (sk, _) = Ed25519.newKeyPair
+
+    (0 until n).forall(i => Validate.blockSignature[Id](signedBlock(i))) should be(true)
+    log.warns should be(Nil)
+  }
+
+  "Block number validation" should "only accept 0 as the number for a block with no parents" in {
+    val chain = createChain[StateWithChain](1).runS(initState).value
+    val block = chain.idToBlocks(0)
+
+    Validate.blockNumber[Id](block.withBlockNumber(1), chain) should be(false)
+    Validate.blockNumber[Id](block, chain) should be(true)
+    log.warns.size should be(1)
+    log.warns.head.contains("not zero, but block has no parents") should be(true)
+  }
+
+  it should "return false for non-sequential numbering" in {
+    val chain = createChain[StateWithChain](2).runS(initState).value
+    val block = chain.idToBlocks(1)
+
+    Validate.blockNumber[Id](block.withBlockNumber(17), chain) should be(false)
+    Validate.blockNumber[Id](block, chain) should be(true)
+    log.warns.size should be(1)
+    log.warns.head.contains("is not one more than parent number") should be(true)
+  }
+
+  it should "return true for sequential numbering" in {
+    val n     = 6
+    val chain = createChain[StateWithChain](n).runS(initState).value
+
+    (0 until n).forall(i => Validate.blockNumber[Id](chain.idToBlocks(i), chain)) should be(true)
+    log.warns should be(Nil)
+  }
+
+  "Parent validation" should "return true for proper justifications and false otherwise" in {
```

Comment:
I've created a bounty task for creating a visualization for this https://rchain.atlassian.net/browse/RHOL-388 . The comments on b7 ~ b9 make it pretty obvious already.